src/databricks/labs/ucx/source_code/redash.py

```diff
@@ -1,24 +1,172 @@
+import logging
+from collections.abc import Iterator
+from dataclasses import replace
+
+from databricks.labs.blueprint.tui import Prompts
+
 from databricks.sdk import WorkspaceClient
-from databricks.sdk.service.sql import Query
+from databricks.sdk.service.sql import Query, Dashboard
+from databricks.sdk.errors.platform import DatabricksError, NotFound
+
+from databricks.labs.ucx.hive_metastore.migration_status import MigrationIndex
+from databricks.labs.ucx.source_code.base import CurrentSessionState
+from databricks.labs.ucx.source_code.queries import FromTable
 
-from databricks.labs.ucx.source_code.base import Fixer
+logger = logging.getLogger(__name__)
 
 
 class Redash:
-    def __init__(self, fixer: Fixer, ws: WorkspaceClient):
-        self._fixer = fixer
+    MIGRATED_TAG = "migrated by UCX"
+    BACKUP_TAG = "backup by UCX"
+
+    def __init__(self, index: MigrationIndex, ws: WorkspaceClient, backup_path: str):
```

Comment:
why don't you depend on `Installation`, so that it's easier to handle all the file placement?

Reply:
rewritten